README.md

```diff
@@ -5,22 +5,55 @@ Add DTrace probes to Diesel connections.
 ## Overview
 
 The `diesel-dtrace` crate provides a diesel [`Connection`][1] that includes DTrace probe points.
-Probes are fired when a connection to the database is established and for each query.
+Probes are fired when a connection to the database is established and for each
+query and transaction.
 
 In order to register probes, your program must invoke `usdt::register_probes()`
 before the probe points are executed.
 
 ## Probes
 
 ```ignore
-diesel-db*:::connection-establish-start(id: uint64_t, conn_id: Uuid, url: &str);
-diesel-db*:::connection-establish-done(id: uint64_t, conn_id: Uuid, success: u8);
-diesel-db*:::query-start(id: uint64_t, conn_id: Uuid, query: &str);
-diesel-db*:::query-done(id: uint64_t, conn_id: Uuid);
-diesel-db*:::transaction-start(id: uint64_t, conn_id: Uuid);
-diesel-db*:::transaction-done(id: uint64_t, conn_id: Uuid);
+/// Fires right before we attempt to establish a connection.
+connection-establish-start(id: &UniqueId, conn_id: Uuid, url: &str)
+/// Fires when we finish establishing a connection, with a flag indicating
+/// whether it succeeded or failed.
+connection-establish-done(id: &UniqueId, conn_id: Uuid, success: u8)
+/// Fires just before issuing a SQL query.
+query-start(id: &UniqueId, conn_id: Uuid, query: &str)
+/// Fires when a query completes.
+query-done(id: &UniqueId, conn_id: Uuid)
+/// Fires when we start a transaction.
+///
+/// This includes the connection ID as well as the depth of the transaction.
+/// As transactions can be nested, _both_ of these are required to unique ID
+/// a transaction in full.
+///
+/// The depth is `0` if there is no outstanding transaction, meaning this is
+/// not nested inside another transaction. Querying the transaction status
+/// may fail, in which case `depth == -1`.
```

Comment:
```suggestion
/// not nested inside another transaction. In the implementation, querying the transaction status
/// may fail, in which case `depth == -1`; this would represent an unknown, internal error.
```

Trying to provide context with regard to what the fuck this might mean.

Reply:
Yeah, that's better thanks.

Reply:
Done in 0358fa4